dev-tools/dep-info.js

```diff
@@ -0,0 +1,123 @@
+/*
+MIT License
+
+Copyright (c) 2021-present, Elastic NV
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+const { join } = require('path');
+const { readFileSync, writeFileSync, readdirSync } = require('fs');
+const { request } = require('undici');
+
+const ROOT_DIR = join(__dirname, '../');
+const ROOT_NODE_MODULES = join(ROOT_DIR, 'node_modules');
+const CHROMIUM_LICENSE_URL =
+  'https://chromium.googlesource.com/chromium/src/+/refs/heads/main/LICENSE?format=TEXT';
+
+function safeReadDirSync(directory) {
+  let files = [];
+  try {
+    files = readdirSync(directory);
+  } catch (_) {
+    // do nothing
```

Comment:
Could we assert the error is of the type(s) we want to drop on the floor, and otherwise throw the error?

```typescript
// something like this, although I didn't actually test this
if (error.text !== 'This is the error I want to ignore') throw error;
```

Reply:
We could just log though, not worth in throwing error when the dep doesnt contain license info in License file or Notice file.

Reply:
++ I think we want it to silently fail here.